Apps/AdminWebClient/src/Server/Services/CommunicationService.cs

```diff
@@ -113,6 +111,31 @@ public RequestResult<IEnumerable<Communication>> GetAll()
             return requestResult;
         }
 
+        /// <inheritdoc />
+        public RequestResult<Communication> Delete(Communication communication)
+        {
+            if (communication.CommunicationStatusCode == CommunicationStatus.Processed)
+            {
+                this.logger.LogError($"Processed communication can't be deleted.");
+                return new RequestResult<Communication>()
+                {
+                    ResultStatus = ResultType.Error,
+                    ResultMessage = "Processed communication can't be deleted.",
+                    ErrorCode = ErrorTranslator.InternalError(ErrorType.InvalidState),
+                };
+            }
+
+            DBResult<Communication> dbResult = this.communicationDelegate.Delete(communication);
+            RequestResult<Communication> result = new RequestResult<Communication>()
+            {
+                ResourcePayload = dbResult.Payload,
+                ResultStatus = dbResult.Status == DBStatusCode.Deleted ? ResultType.Success : ResultType.Error,
+                ResultMessage = dbResult.Message,
+                ErrorCode = dbResult.Status == DBStatusCode.Deleted ? "" : ErrorTranslator.InternalError(ErrorType.CommunicationInternal)
```

Comment:
I don't think is a communication error. It could be concurrency or bad state

Reply:
changed it to Concurrency.

Reply:
updated to "ErrorCode = dbResult.Status == DBStatusCode.Deleted ? string.Empty : ErrorTranslator.ServiceError(ErrorType.CommunicationInternal, ServiceType.Database),"